src/tool/hpcrun/control-knob.c

```diff
@@ -30,20 +34,23 @@ void
 control_knob_init()
 {
   char *s = getenv("HPCRUN_CONTROL_KNOBS");
-  if (s) {
+
 #define INIT_KNOBS(knob_name) \
   control_knobs[knob_name].name = control_knob_names[knob_name]; \
-  control_knobs[knob_name].value = NULL;
+  control_knobs[knob_name].value = default_knob[knob_name];
 
   FORALL_KNOBS(INIT_KNOBS) 
 
 #undef INIT_KNOBS
 
+	if (s) {
```

Comment:
A separate issue: at least one of these control knobs (related to semaphores)  is nvidia specific and shouldn't be available if we are not using an NVIDIA GPU.

Reply:
Yes we have
  macro(**HPCRUN_CUDA_DEVICE_BUFFER_SIZE**, 8388608)     \
  macro(**HPCRUN_CUDA_DEVICE_SEMAPHORE_SIZE**, 65536)

if we don't use Nvidia, these macros won't be used, right, so why we need to hide them?

Reply:
They shouldn't appear as possible knobs unless we are using an nvidia device.